homeassistant/components/iaqualink/__init__.py

```diff
@@ -228,12 +255,12 @@ def unique_id(self) -> str:
     @property
     def assumed_state(self) -> bool:
         """Return whether the state is based on actual reading from the device."""
-        return not self.dev.system.last_run_success
+        return self.dev.system.online in [False, None]
 
     @property
     def available(self) -> bool:
         """Return whether the device is available or not."""
-        return self.dev.system.online
+        return self.dev.system.online is True
```

Comment:
Is `self.dev.system.online` not a bool already?

Reply:
It is... ish. Under some circumstances, the value can be None, which means it's in an unknown state.

Reply:
probably ok to leave as-is then